spec/design/skel/KernelInit_H.thy

```diff
@@ -77,11 +73,19 @@ newKernelState_def:
         ksMachineState = init_machine_state
 \<rparr>"
 
+(* disambiguate newKernelState defined here as global vs the one in Arch as Arch *)
```

Comment:
Very minor, but you seem to have a few different ways of writing this comment for what I think are equivalent cases. For instance, the one here and `(* disambiguate Arch.invocation vs global.invocation *)` in `Invocations_H.thy`. It might be good if they were consistent, to make it obvious in the future that they're handling the same thing.

Reply:
Argh, the hard question. Reading through them again, they actually are for subtly different cases, along with the third phrasing of `(* disambiguate name clash between Arch and non-arch consts with same names *)` which you seem to have used when a collection of consts is being requalified as global at the same time.

I think I am leaning towards that third phrasing as being the most applicable in the different cases, although I'm not that fussed. Consistency would be nice but this isn't worth spending too much time or brain power on optimising.

Reply:
Ok, going with the "third phrasing". Might as well be consistent. I think the first comments were written while I was trying to figure out what was going on.